composer/datasets/streaming/download.py

```diff
@@ -39,8 +39,8 @@ def download_from_s3(remote: str, local: str, timeout: float) -> None:
         timeout (float): How long to wait for shard to download before raising an exception.
     """
     try:
-        import boto3
-        from botocore import Config
+        import boto3  # type: ignore (third-party)
```

Comment:
Makes sense. Instead of the type ignores, thoughts about setting `reportMissingImports="none"` in `pyproject.toml`?

My main hesitation about the "# type: ignore" is that they clutter the code and developers will have to remember to add them in (though, we could run the pyright check on a `dev` install to catch missing type ignores).

That said, changing the setting globally would apply to _all_ imports. I'm not thrilled about that either, since most of the imports should always be installed. This would make typos harder to catch until one actually attempts to import the module (which would happen during tests).

Agree this is an issue, but torn between these options.